DeploymentAgentReconfigurationScripts/TFS2018 Update2/HardwareMoveRemoveOldAgent.ps1

```diff
@@ -0,0 +1,178 @@
+<# 
+This is a guidance script to bring a deployment group agent online after TFS Hardware move activity for TFS2018 Update2. Post Hardware move, all the agents will be offline as the TFS server url has been updated. User can use\follow this script to bring them online. The script has to be run from Administrator PowerShell prompt on the each agent machine.
+
+Inputs:
+1. targetTFSUrl : New TFS url after hardware move.
+2. patToken : PAT token from new TFS with Deployment group manage scope (at least)
+3. agentDownloadUrl: this is optional, user can specify if she wants any specific agent version to be installed.
+4. existingAgentFolder: The script will Auto-detect the agent folder if it was running as windows service. User need to pass the folder path otherwise.
+5. action: By default, the script will not do any update operation, it will just print what changes will happen after the script is applied. User need to set action parameter to 'apply' to execute actual steps.
+
+Output:
+If action is set to apply, this script will
+1. Delete the existing unusable offline agent and re-configure a new agent in with same name and properties including tags.
+2. All existing deployment history will be deleted. 
+
+Example usage:
+.\HardwareMoveRemoveOldAgent.ps1 -targetTFSUrl <newTfsUrl> -patToken <PAT-for-new-TFS>
+.\HardwareMoveRemoveOldAgent.ps1 -targetTFSUrl <newTfsUrl> -patToken <PAT-for-new-TFS> -existingAgentFolder <AgentFoilder (C:\vstsagents\A1)>
+.\HardwareMoveRemoveOldAgent.ps1 -targetTFSUrl <newTfsUrl> -patToken <PAT-for-new-TFS> -existingAgentFolder <AgentFoilder (C:\vstsagents\A1)> -action 'apply'
+#>
+
+param([string]$targetTFSUrl,
+      [string]$patToken,
+      [string]$existingAgentFolder = "",
+      [string]$agentDownloadUrl = 'https://vstsagentpackage.azureedge.net/agent/2.131.0/vsts-agent-win-x64-2.131.0.zip',
```

Comment:
instead of defaulting here, can we take it from const file if not passed

Reply:
we can have const file as per tfs version, we can take/detect the tfs version and as per version we can load the const file

Reply:
I want the script to be single file